features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/factories/event/TimelineItemContentMessageFactory.kt

```diff
@@ -68,12 +69,15 @@ class TimelineItemContentMessageFactory @Inject constructor(
 
     suspend fun create(content: MessageContent, senderDisplayName: String, eventId: EventId?): TimelineItemEventContent {
         return when (val messageType = content.type) {
-            is EmoteMessageType -> TimelineItemEmoteContent(
-                body = "* $senderDisplayName ${messageType.body}",
-                htmlDocument = messageType.formatted?.toHtmlDocument(prefix = "* $senderDisplayName"),
-                formattedBody = parseHtml(messageType.formatted, prefix = "* $senderDisplayName"),
-                isEdited = content.isEdited,
-            )
+            is EmoteMessageType -> {
+                val emoteBody = "* $senderDisplayName ${messageType.body}"
+                TimelineItemEmoteContent(
+                    body = emoteBody,
+                    htmlDocument = messageType.formatted?.toHtmlDocument(prefix = "* $senderDisplayName"),
+                    formattedBody = parseHtml(messageType.formatted, prefix = "* $senderDisplayName") ?: emoteBody.withLinks(),
```

Comment:
If I understand correctly, we always have a `formattedBody` now with those fixes. Could we simplify a bit the `TimelineItemTextBasedContent` interface then? 
It's a bit hard to know what to use currently, and we probably don't need all the fields.
```
sealed interface TimelineItemTextBasedContent : TimelineItemEventContent {
    val body: String
    val htmlDocument: Document?
    val formattedBody: CharSequence?
    val plainText: String
    val isEdited: Boolean
    val htmlBody: String?
        get() = htmlDocument?.body()?.html()
}
```

Reply:
> we always have a `formattedBody`

No, `withLinks()` returns `null` if no links are added.

Reply:
Can we still try to simplify the api? 
If we could get a `plainText/formattedText/body/htmlBody` and remove the `htmlDocument` instance?
I'd say, The `formattedText` shouldn't be nullable and should always be displayed in timeline.

Reply:
Yes, but I would prefer to do that in a separate rework PR. OK for you?

Reply:
as u want